twister: platform filtering when test-only

pavloinfineon opened the pull request:
This PR fixes an issue https://github.com/zephyrproject-rtos/zephyr/issues/62560

- If flags -p and -P are not used twister will use the devices in the testplan.json to figure out what platforms it will run on, **same behavior** as before.
- If flag -P is used it will now exclude that platform from running tests. **Changed behavior**.
- If flag -p is used then only the plaforms defined will have tests run on them. **Changed behavior**.

gchwier commented on scripts/pylib/twister/twisterlib/testplan.py:
```diff
@@ -220,7 +220,12 @@ def load(self):
                     if connected['connected']:
                         connected_list.append(connected['platform'])
-
+            else:
```
`self.options.platform` can be also verified with `self.verify_platforms_existence` method

pavloinfineon committed:
twister: platform filtering when test-only

Add the functionality to use -p and -P to filter out platforms
when using test-only

Signed-off-by: Pavlo Havrylyuk <pavlo.havrylyuk@infineon.com>
